helm/ngrok-operator/templates/agent/service-account.yaml

```diff
@@ -0,0 +1,15 @@
+{{- if .Values.agent.serviceAccount.create -}}
```

Comment:
Q. Do we need to do any backwards compatibility checking for different `.Values` paths for the fields moved under `.Values.agent.*`?

Reply:
The `agent` section is a net new addition to the `values.yaml`. There are only overrides right now for `replicas`, `priorityClassName`, and `serviceAccount` currently. This will likely expand over time as we want to override more settings for the agent. These were chosen to start with since:
* `.Values.agent.serviceAccount.*` - We want to use a more tightly scoped cluster role & binding for this deployment, so we need a new service account. We'll create one by default for the new agent deployment.
* `.Values.agent.priorityClassName` - It seems reasonable that end users might want to give forwarding of traffic to pods a higher priority class than the API reconciliation. This defaults to not being set, but can be changed.
* `.Values.agent.replicas` - This is needed for the migration and also allows users to scale the data forwarding path independent of the api manager.

What kind of backwards compatibility checks are you thinking?